src/thor/route_action.cc

```diff
@@ -469,16 +505,14 @@ void thor_worker_t::path_arrive_by(Api& api, const std::string& costing) {
         auto offset = path.back().elapsed_cost;
         std::for_each(temp_path.begin(), temp_path.end(),
                       [offset](PathInfo& i) { i.elapsed_cost += offset; });
-        // Connects via the same edge so we only need it once
-        if (path.back().edgeid == temp_path.front().edgeid) {
-          path.pop_back();
-        } // Connects on a different edge and is a via location
-        else if (destination->type() == valhalla::Location::kVia) {
-          // Insert a route discontinuity if the paths meet at opposing edges and not
-          // at a graph node. Use path size - 1 as the index where the discontinuity lies.
-          via_discontinuity(*reader, *destination, path.back().edgeid, temp_path.front().edgeid, vias,
-                            temp_path.size(), true);
-        }
+
+        // We should build edge_trimming objects for every via and through
+        // location regardless of what the resulting path is.
+        // Insert a route discontinuity if the paths meet at opposing edges and not
+        // at a graph node. Use path size - 1 as the index where the discontinuity lies.
```

Comment:
Maybe its better here to stop using the nomenclature about discontinuity because its very much untrue after this change